infinispan-remote/src/main/java/org/hibernate/ogm/datastore/infinispanremote/impl/protobuf/schema/SchemaValidator.java

```diff
@@ -92,12 +92,11 @@ private void validate() {
 		}
 	}
 
-	private FileDescriptor parseSchema() {
+	private FileDescriptor parseSchema() throws DescriptorParserException, IOException {
 		FileDescriptorSource fileDescriptorSource = FileDescriptorSource.fromString( schemaName, protoSchema );
 		Configuration config = Configuration.builder().build();
-
-		SquareProtoParser protoParser = new SquareProtoParser( config );
-		Map<String, FileDescriptor> fileDescriptorMap = protoParser.parse( fileDescriptorSource );
```

Comment:
Please note (from parse method javadoc) that semantic checking (it type name resolution) was not previously performed here, but the new code does it. That might lead to strange failures when validating each schema individually as they should be validated together so type resolution can be peformed properly.
I did not look deeply in the how OGM uses schemas to be able to tell if that's going to be an issue.
The test work, but that might not be the whole truth :)

Reply:
The semantic checking is performed by Hibernate OGM, we check that FileDescriptor instance contains all the required caches, messages and fields. Anyway I'm going to add a test on other PR to check the behavior in case of semantic error of provided schema itself.